src/runtime/lpython_builtin.py

```diff
@@ -764,6 +764,14 @@ def _lpython_str_upper(x: str) -> str:
     return res
 
 
+def _lpython_str_join(s:str, lis:list[str]) -> str:
+    res:str = ""
+    i:i32
+    for i in range(len(lis)):
+        res+=s
+        res+=lis[i]
+    return res[1:]
```

Comment:
thats not reliable. You are always assuming your str len will be 1 here, It should fail when str has len more than 1. You can prob use `res[len(str):]`

Reply:
Good catch! I think even better would be something like:
```
if len(lis) == 0: return ""
res = lis[0]
for i in range(1, len(lis)):
    res += s + lis[i]
return res
```

Reply:
That works too, altho as long as there plans of iterables, part of `len` will probably be removed regardless.